### PR TITLE
Fix bug in the frame attributes references for Galactocentric frame

### PR DESCRIPTION
**Pull request by adrn**
### Description
 
While working on astrofrog/astropy#91 to write a what's new entry for the changes to `Galactocentric`, I noticed that the dictionary of references (i.e. scientific papers that the parameter values come from) was not being properly handled. This adjusts the logic to fix how references are stored, and adds a test.

### CODE REVIEW

**Comment by pllim** [comment]
In the spirit of #7928, please run these new tests without ignoring warnings and see if they introduce any new warnings. If so, please handle the warnings accordingly. Thanks!

**Comment by adrn** [comment]
I copied the setup.cfg from #7928 and ran `python setup.py test -t astropy/coordinates` locally -- not seeing any new warnings

**Comment by adrn** [comment]
@pllim Now I did - still no warnings!

**Comment by pllim** [comment]
Thanks for checking, @adrn!

File: astropy/coordinates/tests/test_frames.py [hunk 1]
```diff
@@ -1204,3 +1204,44 @@ def test_galactocentric_default_warning():
...
+        for k in galcen_custom.get_frame_attr_names():
+            if k == 'roll':  # no reference for this parameter
+                continue
+
+            if k == 'z_sun':
+                assert k not in galcen_custom.frame_attribute_references
+            else:
+                assert k in galcen_custom.frame_attribute_references
+
+    reset_galactocentric_defaults()
```
Comment (eteq) [inline astropy/coordinates/tests/test_frames.py]: shouldn't this be in a `finally` block (and then the whole thing in a ``try``)?

Or alternatively, you could use the fixtures finalization magic of pytest by doing something like:
```
@pytest.fixture
def reset_galactocentric_defaults():
    yield 
    ... do what the current reset_galactocentric_defaults does...
```
and then have the pytest call for this function be:
```
def test_galactocentric_references(reset_galactocentric_defaults)
```

File: astropy/coordinates/tests/test_frames.py [hunk 2]
```diff
@@ -1178,16 +1183,14 @@ def test_galactocentric_defaults():
...
 
-def test_galactocentric_default_warning():
+def test_galactocentric_default_warning(reset_galactocentric_defaults):
     from astropy.coordinates import (Galactocentric,
                                      galactocentric_frame_defaults)
 
     # Note: this is necessary because the doctests may alter the state of the
     # class globally and simultaneously when the tests are run in parallel
-    reset_galactocentric_defaults()
+    # reset_galactocentric_defaults()
```
Comment (mhvk) [inline astropy/coordinates/tests/test_frames.py]: Oops, missed in review...

Comment (pllim) [reply astropy/coordinates/tests/test_frames.py]: So is this line supposed to be removed?

Comment (mhvk) [reply astropy/coordinates/tests/test_frames.py]: Indeed, no reason to just comment it out, it is now done in `setup` and `teardown`. Indeed, the comment above can go too...

Comment (pllim) [reply astropy/coordinates/tests/test_frames.py]: OK, I'll fold it into #7928.

**Pull request merged**